Mu2eG4/src/constructDS.cc

```diff
@@ -174,23 +174,41 @@ namespace mu2e {
 
     // - upstream face
     GeomHandle<Beamline> beamg;
-    const StraightSection * ts5out = beamg->getTS().getTSCryo<StraightSection>( TransportSolenoid::TSRegion::TS5,
+    auto ts(&beamg->getTS());
+    const StraightSection * ts5out = ts->getTSCryo<StraightSection>( TransportSolenoid::TSRegion::TS5,
                                                                                 TransportSolenoid::TSRadialPart::OUT );
 
+    double interconnectHL = _config.getDouble("ds.interconnect.halfLength", -1.);
     double dsFrontZ0 = dsP.z() - ds->halfLength() + ds->frontHalfLength();
-    TubsParams    dsFrontParams  ( ts5out->rOut(), dsInnerCryoParams.innerRadius(), ds->frontHalfLength() );
-    G4ThreeVector dsFrontPosition( dsP.x(), dsP.y(), dsFrontZ0);
-
-    nestTubs( "DSFront",
-              dsFrontParams,
-              dsCryoMaterial,
-              0,
-              dsFrontPosition-_hallOriginInMu2e,
-              parent,
-              0,
-              G4Color::Blue(),
-              "ds"
-              );
+    double smallOffset = 1.e-3; //create small buffers to prevent overlaps on both ends of the interconnect
+    if(interconnectHL < 0.) {
+      TubsParams    dsFrontParams  ( ts5out->rOut(), dsInnerCryoParams.innerRadius(), ds->frontHalfLength() );
+      G4ThreeVector dsFrontPosition( dsP.x(), dsP.y(), dsFrontZ0);
+
+      nestTubs( "DSFront",
+		dsFrontParams,
+		dsCryoMaterial,
+		0,
+		dsFrontPosition-_hallOriginInMu2e,
+		parent,
+		0,
+		G4Color::Blue(),
+		"ds"
+		);
+    } else {
+      double interconnectRIn  = _config.getDouble("ds.interconnect.rIn");
```

Comment:
For future reference.   In most circumstances I prefer of a pattern in which the  else {} catches an illegal configuration rather than defaulting to the last legal one.  If, in the future, you are adding a new version and forget to add an extra else if, then this catches it.

Reply:
That makes sense, I'll try to follow this style in the future!